docs/development/pr-review.asciidoc

```diff
@@ -0,0 +1,129 @@
+
+On the other hand, if a reviewer calls out a specific chunk of code as being hard to understand, then one of the first people to look at the code is having a hard time understanding it. If the reviewer feels strongly that this is a problem, they should propose alternatives for the author's consideration.
+
+When in doubt, relying on "prior art" in the codebase, especially in and around the area of the codebase that is being contributed to, is generally a good idea.
+
+There may also be times when a person is inspired by a particular contribution to introduce a new way to style code that we already have different style guidelines or "prior art" for. It's OK to bring this up in a pull request, but ultimately that discussion should branch off into a separate issue or pull request to update the appropriate guide. If this change is prompted by a reviewer, then the original PR should not be blocked on this.  If the change is prompted by the author, then they can either update the PR to be consistent with our existing guidelines (preferred) or they can choose to block the PR entirely on that separate styleguide discussion.
+
+
+[float]
+=== Nitpicking
+
+Nitpicking is when a reviewer identifies trivial and unimportant details in a pull request and asks the author to change them. This is a completely subjective category that is impossible to define universally, and it's equally impractical to define a blanket policy on nitpicking that everyone will be happy with.
+
+Reviewers should feel comfortable giving any feedback they have on a pull request regardless of how trivial it is. Authors should feel equally comfortable passing on feedback that they think is trivial and inconsequential.
+
+Often, reviewers have an opinion about whether the feedback they are about to give is a nitpick or not. While not required, it can be really helpful to identify that feedback as such, for example "nit: a newline after this would be helpful". This helps the author understand your intention.
+
+
+[float]
+=== Review feedback that is never appropriate
+
+Whether or not a bit of feedback is appropriate for a pull request is really dependent on the motivation for giving the feedback in the first place. Reviewers should not be reviewing code with the mindset of "how would I write this code?". The reviewer didn't write the code, and their goal isn't to craft the contribution into a form that is simply whatever they would have written if they had.
```

Comment:
A bit hard to distinguish this from the above section on nits.  Isn't that what nits are really?  Personal preferences the reviewee can take or leave?  Putting it as acceptable above then here saying that feedback is never appropriate seems a bit confusing.  I'd maybe just leave this section out, or put it above as, "try not to take this attitude in your nits".  then keep this section for stuff like, be supportive, don't be mean.

Reply:
Let me ponder a different way to phrase this that doesn't cover nitpicks. When I wrote this, I was thinking about bigger disagreements like "you should do this with redux instead".

Reply:
Blegh, even that comment doesn't really represent the type of feedback I'm thinking about.

Reply:
Never is a strong word... like, if you do this, you are in big trouble! :)  and it does seem appropriate to put a "don't be mean" clause under here, but if a reviewer isn't quite sure if their feedback falls into this category or not... I'd be worried they would not say something relevant because of the strong wording.  Like I mistook it to mean nits - e.g. I might suggest in a code review, hey you can make this  chunk of code shorter:
```
if (x) {
 return 'hi';
} else {
 return 'bye'
}
```
by doing
```
return x ? 'hi' : 'bye';
```

Maybe that doesn't fall into that category because I can give a reason why I would write it that way - it's fewer lines of code.  

Perhaps the distinction is - can you back up your feedback with something more concrete than "I like it better this way"?  Even so... a comment like "nit: I think this way is more readable" is very close to being the same as "nit: I like it better written this way". 🤷‍♀️

Reply:
@Stacey-Gammon shouldn't the style guideline be the authority in your example. A while ago we decided to adopt airbnb style guidelines (at least most of it I think). If that's the case, maybe a clarification that a deviation from the style guidelines justifies a review comment?

Reply:
I can't actually find a rule about that in the airbnb style guide... to either prefer one or the other format.

Though regardless of my overly simplistic example, there are other situations where the format of code is too complex to be in a style guide.  Deciding between two code snippets where one is shorter and the other (might be) easier to read is a subjective judgement call, and something I might call out in a review, but can't always be boiled down to a style guide rule.

Reply:
> I can't actually find a rule about that in the airbnb style guide... to either prefer one or the other format.

sure, and nothing prevents us from extending the guideline further (I'd +1 a rule that covers the use case you're describing :))

Reply:
I rewrote the bulk of this section to hopefully be more clear. The idea isn't that this feedback is universally inappropriate, but refusing to move a PR forward due to this class of feedback is.